libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java

```diff
@@ -62,6 +75,30 @@ public boolean isUserOffRoute(Location location, RouteProgress routeProgress,
   }
 
   /**
+   * Method to check if the user has passed either the set (in {@link MapboxNavigationOptions})
+   * minimum amount of seconds or minimum amount of meters since the last reroute.
+   * <p>
+   * If the user is above both thresholds, then the off-route can proceed.  Otherwise, ignore.
+   *
+   * @param location current location from engine
+   * @param options  for second (default 3) / distance (default 50m) minimums
+   * @return true if valid, false if not
+   */
+  private boolean validOffRoute(Location location, MapboxNavigationOptions options) {
+    // Check if minimum amount of time has passed since last reroute
+    long secondsBeforeRerouteAllowed = TimeUnit.SECONDS.toMillis(options.secondsBeforeReroute());
+    if (location.getTime() > timeSinceLastOffRoute + secondsBeforeRerouteAllowed) {
+      timeSinceLastOffRoute = location.getTime();
+      return true;
```

Comment:
Let's remove this 3 second check for now and just go with the minimumDistance check for consistency with iOS. 

Thinking more about this, 25/50m isn't great for slow speed walking (that's 40s), but we can revisit this cross-platform in another PR.

cc @bsudekum

Reply:
Would it make sense to adjust the minimumDistance based on directions profile?

Reply:
Yeah, I've noted this to revisit.